### PR TITLE
feat(mobile): exclude locales from overpass font

### PR DESCRIPTION
**Pull request by johnstef99**
Fixes #10436 

As mentioned [here](https://github.com/immich-app/immich/issues/10436#issuecomment-2429775832) this PR adds a list of languages that should be excluded from using Overpass as a font. These languages will use the default Flutter font based on the platform.

As of right now the only known to me language having a problem is Greek, feel free to exclude more if needed.

### CODE REVIEW

File: mobile/lib/main.dart [hunk 1]
```diff
@@ -212,20 +213,31 @@ class ImmichAppState extends ConsumerState<ImmichApp>
...
-        debugShowCheckedModeBanner: false,
-        themeMode: ref.watch(immichThemeModeProvider),
-        darkTheme: getThemeData(colorScheme: immichTheme.dark),
-        theme: getThemeData(colorScheme: immichTheme.light),
-        routeInformationParser: router.defaultRouteParser(),
-        routerDelegate: router.delegate(
-          navigatorObservers: () => [TabNavigationObserver(ref: ref)],
+    return ProviderScope(
+      overrides: [
+        localeProvider.overrideWithValue(context.locale),
```
Comment (alextran1502) [inline mobile/lib/main.dart:218]: What is the purpose of this `overrides` property?

Comment (johnstef99) [reply mobile/lib/main.dart:218]: It sets the `localeProvider` value for the rest subtree of widgets every time that `context.locale` gets updated. Do you have an alternative way of settings the initial value of `localeProvider` and then updating it?

**Comment by VelocityRa** [comment]
> As of right now the only known to me language having a problem is Greek, feel free to exclude more if needed.

Serbian Cyrillic has the issue too: https://github.com/immich-app/immich/issues/11216

**Comment by johnstef99** [comment]
> > As of right now the only known to me language having a problem is Greek, feel free to exclude more if needed.
> 
> Serbian Cyrillic has the issue too: #11216, could be added now. Might also want to tweak the filtering to take the country code into account too - Serbian Cyrillic is specifically `Locale('sr', 'Cyrl')`, there's Serbian Latin too which presumably renders fine (I asked on the issue to make sure).

Ok I will update it to filter with region/country subtag and exclude `Locale('sr', 'Cyrl')`

**Comment by alextran1502** [comment]
The `localProvider` isn't used correctly here, as the value never updates. The theme is already initialized when the locale changes, so you don't need the provider.

You can achieve the same effect by removing the `localProvider` across the app then changing the `getThemeData` method signature to the following

image

**Comment by johnstef99** [comment]
Have you tested that the localeProvider's value never changes? Because I can't reproduce it. When the EasyLocalization parent widget rebuilds the localeProvider's value gets updated. 

Let me explain why I added this provider:

1. In the `MapThemeOveride` widget we also need the locale to get the theme data. If `conext.locale` was used here, it would be difficult for the widget to be tested (and there are already written tests for this widget). The already written test will have to be refactored to have as parent widget the EasyLocalization.
2. I have opened [an other MR](https://github.com/immich-app/immich/pull/14089#issue-2650586726) that also needs the `localeProvider` for the reason that some UI elements are cached by providers. That results on not refreshing the UI after the user changing language.

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**